### PR TITLE
Add firmware variant ZigbeeBridge

### PR DESCRIPTION
**Pull request by Jason2866**
Since @s-hadinger did bring the adventures with the ZigbeeBridge to a sucess it is time for
a firmware variant for.  Surprise we call it `tasmota-zbbridge` 

## Checklist:
  - [x] The pull request is done against the latest dev branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR.
  - [x] The code change is tested and works on Tasmota core ESP8266 V.2.7.2.1
  - [x] The code change is tested and works on core ESP32 V.1.12.2
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/tasmota_configurations.h [hunk 1]
```diff
@@ -406,6 +406,264 @@
...
+#define USE_RULES                                // Add support for rules (+4k4 code)
+//  #define USE_EXPRESSION                         // Add support for expression evaluation in rules (+3k2 code, +64 bytes mem)
+//    #define SUPPORT_IF_STATEMENT                 // Add support for IF statement in rules (+4k2 code, -332 bytes mem)
+//  #define SUPPORT_MQTT_EVENT                     // Support trigger event with MQTT subscriptions (+3k5 code)
+
+//#define USE_SCRIPT                               // Add support for script
+  #define USE_SCRIPT_FATFS 4                     // Add support for script storage on SD card (+12k code, +4k mem)
+
+// -- Optional modules ----------------------------
+//#define ROTARY_V1                                // Add support for MI Desk Lamp
```
Comment (arendst) [inline tasmota/tasmota_configurations.h:454]: As this is enabled in my_user_cfg.h shouldn't they all be undef?

**Comment by meingraham** [comment]
Are there other device specific variants? For example, Sonoff RF Bridge doesn't have its own variant; does it? Is this the right path to take for this? Pandora's Box?

**Comment by arendst** [comment]
As the zigbee code takes up so much of the total code space available it's imposible to have zigbee in the default tasmota.bin. Other devices still fit in tasmota.bin so I'm afraid we have to choose this way. I rather would see ALL zigbee code in one binary though...

**Comment by meingraham** [comment]
I think that's the direction that the variant should take rather than device specific. Like there is an IR variant, there can be a Zigbee variant... which includes the Z2T and Sonoff ZBridge code.

**Comment by Jason2866** [comment]
@arendst  you are right. I have the undef in my override file. Will add

File: tasmota/tasmota_configurations.h (hunk 1 above)
Comment (Jason2866) [reply tasmota/tasmota_configurations.h]: right, changed

**Comment by Jason2866** [comment]
@meingraham  Let us do in steps. Get a precompiled working version first. Adding a (maybe possible) universal zigbee version in second step. At the moment there is no precompiled zigbee version. So imho this is better than having nothing.

**Pull request merged**